src/main/java/eu/eutampieri/catacombs/model/Boss.java

```diff
@@ -60,15 +68,30 @@ public Boss(final int x, final int y, final TileMap tileMap) {
                 changeDirection();
             }
         }
-
+        if (others.stream().filter((x) -> x instanceof Player)
+                .findFirst()
+                .get()
+                .getHitBox()
+                .overlaps(this.radarBox) && this.weapon.canFire) {
```

Comment:
Maybe we could first filter with the overlaps condition and then call findFirst.
Consider that a minor refactoring would be needed because then you could get a null player.

Reply:
So we iterate the player search in the boss.overlaps? 
if so we might do it for every entity.

Reply:
```java
        others.stream().filter((x) -> x instanceof Player)
                .filter((x) -> x.getHitBox()
                        .overlaps(this.radarBox))
                .findFirst()
                .ifPresentOrElse((x) -> { 
                    if (this.weapon.canFire()) { 
                        setShootingDirection(x); 
                    } 
                    }, () -> this.weapon.setCanFire(false));
```
This way